py-polars/src/lazy/dsl.rs

```diff
@@ -562,28 +562,33 @@ impl PyExpr {
         exact: bool,
         cache: bool,
         tz_aware: bool,
+        tu: Option<Wrap<TimeUnit>>,
     ) -> PyExpr {
-        let tu = match fmt {
-            Some(ref fmt) => {
-                if fmt.contains("%.9f")
-                    || fmt.contains("%9f")
-                    || fmt.contains("%f")
-                    || fmt.contains("%.f")
-                {
-                    TimeUnit::Nanoseconds
-                } else if fmt.contains("%.3f") || fmt.contains("%3f") {
-                    TimeUnit::Milliseconds
-                } else {
-                    TimeUnit::Microseconds
+        let result_tu = if let Some(value) = tu {
+            value.0
+        } else {
+            match fmt {
+                Some(ref fmt) => {
+                    if fmt.contains("%.9f")
+                        || fmt.contains("%9f")
+                        || fmt.contains("%f")
+                        || fmt.contains("%.f")
+                    {
+                        TimeUnit::Nanoseconds
+                    } else if fmt.contains("%.3f") || fmt.contains("%3f") {
+                        TimeUnit::Milliseconds
+                    } else {
+                        TimeUnit::Microseconds
+                    }
                 }
+                None => TimeUnit::Microseconds,
             }
-            None => TimeUnit::Microseconds,
```

Comment:
I wanted to try to write ths as
```
        let result_tu = match (fmt, tu) {
            (_, Some(tu)) => tu.0,
            (Some(ref fmt), None) => {
                if fmt.contains("%.9f")
                    || fmt.contains("%9f")
                    || fmt.contains("%f")
                    || fmt.contains("%.f")
                {
                    TimeUnit::Nanoseconds
                } else if fmt.contains("%.3f") || fmt.contains("%3f") {
                    TimeUnit::Milliseconds
                } else {
                    TimeUnit::Microseconds
                }
            }
            (None, None) => TimeUnit::Microseconds,
        };
```

but then I get

```
   Compiling py-polars v0.15.14 (/home/marcogorelli/polars-dev/py-polars)
error[E0382]: use of moved value: `fmt`] 334/335: py-polars                                                                                    
   --> src/lazy/dsl.rs:589:17
    |
560 |         fmt: Option<String>,
    |         --- move occurs because `fmt` has type `std::option::Option<std::string::String>`, which does not implement the `Copy` trait
...
567 |         let result_tu = match (fmt, tu) {
    |                                --- value moved here
...
589 |                 fmt,
    |                 ^^^ value used here after move


error: aborting due to previous error> ] 334/335: py-polars                                                                                    


For more information about this error, try `rustc --explain E0382`.

```